mzLib/Readers/Thermo/ThermoRawFileReader.cs

```diff
@@ -52,49 +49,49 @@ public override MsDataFile LoadAllStaticData(FilteringParams filteringParams = n
             // I don't know why this line needs to be here, but it does...
             var temp = RawFileReaderAdapter.FileFactory(FilePath);
 
-            var threadManager = RawFileReaderFactory.CreateThreadManager(FilePath);
-            var rawFileAccessor = threadManager.CreateThreadAccessor();
-
-            if (!rawFileAccessor.IsOpen)
+            using (var threadManager = RawFileReaderFactory.CreateThreadManager(FilePath))
             {
-                throw new MzLibException("Unable to access RAW file!");
-            }
+                var rawFileAccessor = threadManager.CreateThreadAccessor();
 
-            if (rawFileAccessor.IsError)
-            {
-                throw new MzLibException("Error opening RAW file!");
-            }
+                if (rawFileAccessor.IsError)
+                {
+                    throw new MzLibException("Error opening RAW file!");
+                }
 
-            if (rawFileAccessor.InAcquisition)
-            {
-                throw new MzLibException("RAW file still being acquired!");
-            }
+                if (!rawFileAccessor.IsOpen)
+                {
+                    throw new MzLibException("Unable to access RAW file!");
+                }
 
-            rawFileAccessor.SelectInstrument(Device.MS, 1);
-            var msDataScans = new MsDataScan[rawFileAccessor.RunHeaderEx.LastSpectrum];
+                if (rawFileAccessor.InAcquisition)
+                {
+                    throw new MzLibException("RAW file still being acquired!");
+                }
 
-            Parallel.ForEach(Partitioner.Create(0, msDataScans.Length), new ParallelOptions { MaxDegreeOfParallelism = maxThreads }, (fff, loopState) =>
-            {
-                IRawDataPlus myThreadDataReader = threadManager.CreateThreadAccessor();
```

Comment:
i don't think these should be moved out of the loop. each parallel loop may need its own to prevent collions.

Reply:
I think they still exist in the loop at line 77. They're instantiated inside a using statement